src/algebra/group_power/basic.lean

```diff
@@ -142,6 +142,12 @@ by rw [pow_bit0, (commute.refl a).mul_pow]
 theorem pow_bit1' (a : M) (n : ℕ) : a ^ bit1 n = (a * a) ^ n * a :=
 by rw [bit1, pow_succ', pow_bit0']
 
+@[to_additive]
+lemma pow_mul_pow_eq_one {a b : M} : ∀ (n : ℕ) (h : a * b = 1), a ^ n * b ^ n = 1
+| 0 h := by rw [pow_zero, pow_zero, mul_one]
+| (n+1) h := by rw [pow_add, add_comm, pow_add, pow_one, pow_one, mul_assoc, ← mul_assoc a, h,
+  one_mul, pow_mul_pow_eq_one n h]
```

Comment:
```suggestion
lemma pow_mul_pow_eq_one {a b : M} (n : ℕ) (h : a * b = 1) :
  a ^ n * b ^ n = 1 := 
begin
  induction n with n hn,
  { simp },
  { calc a ^ n.succ * b ^ n.succ = a ^ n * a * (b * b ^ n) : by rw [pow_succ', pow_succ]
    ... = a ^ n * (a * b) * b ^ n : by group
    ... = a ^ n * b ^ n : by simp [h]
    ... = 1 : hn }
end
```
It's better to don't use `∀` after the colon.

Reply:
Ops, for you `M` is only a monoid (as it should be), so the line `by group` must be done by hand, but it should be only a bunch of `mul_assoc`.